core/src/main/java/com/uber/m3/tally/CounterImpl.java

```diff
@@ -27,15 +27,24 @@
 /**
  * Default implementation of a {@link Counter}.
  */
-class CounterImpl implements Counter {
+class CounterImpl extends MetricBase implements Counter {
     private AtomicLong prev = new AtomicLong(0);
     private AtomicLong curr = new AtomicLong(0);
 
+    protected CounterImpl(String fqn) {
+        super(fqn);
+    }
+
     @Override
     public void inc(long delta) {
         curr.getAndAdd(delta);
     }
 
+    @Override
```

Comment:
I don't live in Java code so this more a curiosity: why override this method if you're going to only call the parent's method? Is it required by the Java type system?

Reply:
This is a refactoring aberration, cleaned up in #74